### PR TITLE
softmax cross entropy without reduction

### PR DESCRIPTION
**Pull request by delta2323**
This PR removes reduce option from `SoftmaxCrossEntropy` and `softmax_cross_entropy`. Different from v1, they output elementwise loss values in v2. Related to #2558 #2357

### CODE REVIEW

**Comment by delta2323** [comment]
Should we change `L.Classifier` from
```
self.loss = self.lossfun(self.y, t)
```
to
```
self.loss = F.sum(self.lossfun(self.y, t))
```
because the default value of `lossfun` is `F.softmax_cross_entropy`

**Comment by delta2323** [comment]
Following the tentative conclusion I wrote [here](https://github.com/pfnet/chainer/issues/2558#issuecomment-300378788), I close this PR. Thank you.
